### PR TITLE
Improve OpenGL context creation

### PR DESCRIPTION
**Pull request by binary1248**
* Make Glx context creation favour the attribsARB variant if it is available, just like how it is already done in the Wgl context creation.
* Interpret context version settings <= 1.1 as requesting the most recent context version supported by the driver. This means the user will get the most recent version if they don't provide their own settings. There aren't any disadvantages to doing this.

### CODE REVIEW

File: src/SFML/Window/Unix/GlxContext.cpp [hunk 1]
```diff
@@ -486,21 +479,18 @@ void GlxContext::createContext(GlxContext* shared, unsigned int bitsPerPixel, co
...
-                // Create the context
-                int attributes[] =
-                {
-                    GLX_CONTEXT_MAJOR_VERSION_ARB, static_cast<int>(m_settings.majorVersion),
-                    GLX_CONTEXT_MINOR_VERSION_ARB, static_cast<int>(m_settings.minorVersion),
-                    0,                             0
-                };
+            // Append the terminating 0
+            attributes.push_back(0);
+            attributes.push_back(0);
```
Comment (mantognini) [inline src/SFML/Window/Unix/GlxContext.cpp]: Is it intentional there're two terminating `0`?

File: src/SFML/Window/Win32/WglContext.cpp [hunk 2]
```diff
@@ -426,15 +433,14 @@ void WglContext::createContext(WglContext* shared, unsigned int bitsPerPixel, co
...
-                    WGL_CONTEXT_MAJOR_VERSION_ARB, static_cast<int>(m_settings.majorVersion),
-                    WGL_CONTEXT_MINOR_VERSION_ARB, static_cast<int>(m_settings.minorVersion),
-                    0,                             0
-                };
-                m_context = wglCreateContextAttribsARB(m_deviceContext, sharedContext, attributes);
             }
+
+            // Append the terminating 0
+            attributes.push_back(0);
+            attributes.push_back(0);
```
Comment (mantognini) [inline src/SFML/Window/Win32/WglContext.cpp]: Same question.

Comment (binary1248) [inline src/SFML/Window/Win32/WglContext.cpp]: The attributes are read in pairs... so theoretically the operating system doesn't have to read the second one, but who knows if it really does... There are a lot of broken things in X.

Comment (mantognini) [inline src/SFML/Window/Win32/WglContext.cpp]: Alright. I didn't follow this policy on OS X though, for what it's worth.

File: src/SFML/Window/Unix/GlxContext.cpp (hunk 1 above)
Comment (minirop) [inline src/SFML/Window/Unix/GlxContext.cpp]: `glXCreateContextAttribsARB` is probably reading values 2 by 2.

Comment (mantognini) [inline src/SFML/Window/Unix/GlxContext.cpp]: > edit: the doc uses only 1 zero.

good to know. :-)

**Comment by binary1248** [comment]
Bump.

**Comment by eXpl0it3r** [comment]
This PR has been added to my merge list, meaning it will be merged soon, unless someone raises any concerns.

**Pull request merged**